### PR TITLE
8285951: Replace Algorithms.eatMemory(...) with WB.fullGC() in vmTestbase_vm_gc_ref tests

### PR DESCRIPTION
**Pull request by lmesnik**
remainging weak/soft/phantom reference tests are updated to use WB.fullGC()

The TEST.properties are removed for single-threaded tests.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Issue
 * [JDK-8285951](https://bugs.openjdk.java.net/browse/JDK-8285951): Replace Algorithms.eatMemory(...) with WB.fullGC() in vmTestbase_vm_gc_ref tests

### Reviewers
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**)
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8474/head:pull/8474` \
`$ git checkout pull/8474`

Update a local copy of the PR: \
`$ git checkout pull/8474` \
`$ git pull https://git.openjdk.java.net/jdk pull/8474/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8474`

View PR using the GUI difftool: \
`$ git pr show -t 8474`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8474.diff">https://git.openjdk.java.net/jdk/pull/8474.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/gc/gctests/PhantomReference/phantom001/phantom001.java [hunk 1]
```diff
@@ -54,22 +53,23 @@
  *
```
Comment (kimbarrett) [inline test/hotspot/jtreg/vmTestbase/gc/gctests/PhantomReference/phantom001/phantom001.java:53]: Pre-existing: There doesn't seem to be any code for check (6) - reference.clear() doesn't throw.  You fixed this in some of the tests, but not this one.

**Review by kimbarrett — APPROVED** [review]
Looks good.

File: test/hotspot/jtreg/vmTestbase/gc/gctests/PhantomReference/phantom001/phantom001.java (hunk 1 above)
Comment (lmesnik) [reply test/hotspot/jtreg/vmTestbase/gc/gctests/PhantomReference/phantom001/phantom001.java:52]: I removed this comment. We don't test clear().

**Review by tschatzl — APPROVED** [review]
Seems good. It seems that with this change we can close out all existing issues with these tests (phantom*, soft*, etc) that fail because of intermittent OOME.

**Review by kimbarrett — APPROVED** [review]
Still looks good.

**Pull request closed without merging**